stdlib/public/Cxx/CxxSet.swift

```diff
@@ -13,11 +13,31 @@
 public protocol CxxSet<Element> {
   associatedtype Element
   associatedtype Size: BinaryInteger
+  associatedtype InsertionResult // std::pair<iterator, bool>
+
+  init()
+
+  @discardableResult
+  mutating func __insertUnsafe(_ element: Element) -> InsertionResult
 
   func count(_ element: Element) -> Size
 }
 
 extension CxxSet {
+  /// Creates a C++ set containing the elements of a Swift Sequence.
+  ///
+  /// This initializes the set by copying every element of the sequence.
+  ///
+  /// - Complexity: O(*n*), where *n* is the number of elements in the Swift
+  ///   sequence
+  @inlinable
+  public init<S: Sequence>(_ sequence: S) where S.Element == Element {
```

Comment:
shouldn't this be `__shared S`? we want to borrow it right?

Reply:
Good point, I'll put up a patch for main branch first, then will add it here as well